### ISSUE TITLE
Exception : Children of TabBar must have the tab role in 3.32.0

### ISSUE THREAD

**Issue by scolnet** [flutter/flutter#169175]
### Steps to reproduce

After upgrading 3.32 and using TabBarView

### Expected results

No exception

### Actual results

Exception caught by scheduler library when tabbarview is displayed. 

Error appear only on IOS simulator.
It's ok on Android and Web

The same code run perfectly in version 3.29.3

### Code sample

<details open><summary>Code sample</summary>

```dart
[Paste your code here]
```

</details>

### Screenshots or Video

<details open>
<summary>Screenshots / Video demonstration</summary>

[Upload media here]

</details>

### Logs

<details open><summary>Logs</summary>

```console
`The following assertion was thrown during a scheduler callback:
Children of TabBar must have the tab role

When the exception was thrown, this was the stack: 
SemanticsNode._addToUpdate.<anonymous closure> (package:flutter/src/semantics/semantics.dart:3467:9)
SemanticsNode._addToUpdate (package:flutter/src/semantics/semantics.dart:3470:6)`
```

</details>

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
[Paste your output here]
```

</details>

**Comment by acfsmith** [comment]
Linux version is also affected by the problem.

**Comment by dattm2593** [comment]
Same issue here, I'm using `go_router`

`DefaultTabController(
      length: NavBarType.values.length,
      child: BlocConsumer<BottomNavigationCubit, BottomNavigationState>(
        listenWhen: (pre, cur) {
          return pre.navBarType != cur.navBarType;
        },
        listener: (context, state) {
          final index = NavBarType.values.indexOf(state.navBarType);
          if (index == -1) return;
          navigationShell.goBranch(
            index,
            initialLocation: index == navigationShell.currentIndex,
          );
          DefaultTabController.of(context).animateTo(index);
        },
        builder: (context, state) {
          final navBarType = state.navBarType;

          return Scaffold(
            body: navigationShell,
            bottomNavigationBar: TabBar(
              indicator: TopIndicator(AppColors.blue),
              unselectedLabelStyle: AppTextStyles.unSelectedNavBar,
              labelStyle: AppTextStyles.selectedNavBar,
              unselectedLabelColor: AppColors.unselectedText,
              labelColor: AppColors.blue,
              indicatorSize: TabBarIndicatorSize.tab,
              tabs: _tabBarItems(context, navBarType),
              onTap: (i) => _onItemTapped(i, context),
            ),
          );
        },
      ),
    );`

**Comment by VB10** [comment]
I've similar issue too. I search about this problem and i've seen this PR. May i think after this this problem start.
@kevmoo @chunhtai 
https://github.com/flutter/flutter/pull/161260

**Comment by darshankawar** [comment]
> [@kevmoo](https://github.com/kevmoo) [@chunhtai](https://github.com/chunhtai)
> [#161260](https://github.com/flutter/flutter/pull/161260)

@VB10 Please don't directly tag Flutter team members for attention. There's a triage process in place for all issues.

**Comment by scolnet** [comment]
@darshankawar 
After several tests : 

Exception is thrown when you use a widget in tab which doesn't have semantic label by default.
For exemple Image.asset or Image.network 

if you try your code and change tab like this, you would have an exception : 
```dart
bottom: TabBar(
            tabs: <Widget>[
              Tab(icon: Image.network('https://storage.googleapis.com/cms-storage-bucket/3461c6a5b33c339001c5.jpg')),
              Tab(icon: Icon(Icons.beach_access_sharp)),
              Tab(icon: Icon(Icons.brightness_5_sharp)),
            ],
          ),
``` 

Workaround : 
Add semanticLabel for the widget like this : 
```dart
 Image.network('https://storage.googleapis.com/cms-storage-bucket/3461c6a5b33c339001c5.jpg'),  semanticLabel:'Tab 1')
``` 

If I understand the need of accessibility, it would not be mandatory.

**Comment by chunhtai** [comment]
I can reproduce this, it looks like there is a frame or two before the image is loaded  that somehow creates an empty node.

This is a bug should be fixed

**Comment by chunhtai** [comment]
mark as p0 since this is a regression that may cause a lot of customer breakage.

For those who are facing this issue, these assert are safe to ignore and your app should runs fine under release mode

**Comment by darshankawar** [comment]
Verified per below updated code sample which throws the reported error, while the same works properly in previous stable.

<details>
<summary> code </summary>

```

import 'package:flutter/material.dart';

/// Flutter code sample for [TabBar].

void main() => runApp(const TabBarApp());

class TabBarApp extends StatelessWidget {
  const TabBarApp({super.key});

  @override
  Widget build(BuildContext context) {
    return const MaterialApp(home: TabBarExample());
  }
}

class TabBarExample extends StatelessWidget {
  const TabBarExample({super.key});

  @override
  Widget build(BuildContext context) {
    return DefaultTabController(
      initialIndex: 1,
      length: 3,
      child: Scaffold(
        appBar: AppBar(
          title: const Text('TabBar Sample'),
          bottom: TabBar(
            tabs: <Widget>[
              Tab(icon: Image.network('https://storage.googleapis.com/cms-storage-bucket/3461c6a5b33c339001c5.jpg')),
              Tab(icon: Icon(Icons.beach_access_sharp)),
              Tab(icon: Icon(Icons.brightness_5_sharp)),
            ],
          ),
        ),
        body: const TabBarView(
          children: <Widget>[
            Center(child: Text("It's cloudy here")),
            Center(child: Text("It's rainy here")),
            Center(child: Text("It's sunny here")),
          ],
        ),
      ),
    );
  }
}

```

</details>

<details>
<summary> error log </summary>

```

======== Exception caught by scheduler library =====================================================
The following assertion was thrown during a scheduler callback:
Children of TabBar must have the tab role

When the exception was thrown, this was the stack: 
#0      SemanticsNode._addToUpdate.<anonymous closure> (package:flutter/src/semantics/semantics.dart:3467:9)
#1      SemanticsNode._addToUpdate (package:flutter/src/semantics/semantics.dart:3470:6)
#2      SemanticsOwner.sendSemanticsUpdate (package:flutter/src/semantics/semantics.dart:4233:14)
#3      PipelineOwner.flushSemantics (package:flutter/src/rendering/object.dart:1502:24)
#4      PipelineOwner.flushSemantics (package:flutter/src/rendering/object.dart:1504:15)
#5      RendererBinding.drawFrame (package:flutter/src/rendering/binding.dart:636:25)
#6      WidgetsBinding.drawFrame (package:flutter/src/widgets/binding.dart:1242:13)
#7      RendererBinding._handlePersistentFrameCallback (package:flutter/src/rendering/binding.dart:495:5)
#8      SchedulerBinding._invokeFrameCallback (package:flutter/src/scheduler/binding.dart:1438:15)
#9      SchedulerBinding.handleDrawFrame (package:flutter/src/scheduler/binding.dart:1351:9)
#10     SchedulerBinding.scheduleWarmUpFrame.<anonymous closure> (package:flutter/src/scheduler/binding.dart:1060:9)
#11     PlatformDispatcher.scheduleWarmUpFrame.<anonymous closure> (dart:ui/platform_dispatcher.dart:889:16)
#15     _RawReceivePort._handleMessage (dart:isolate-patch/isolate_patch.dart:193:12)
(elided 3 frames from class _Timer and dart:async-patch)
====================================================================================================

======== Exception caught by Flutter framework =====================================================
The following assertion was thrown while executing callbacks for FrameTiming:
'package:flutter/src/widgets/binding.dart': Failed assertion: line 1218 pos 16: 'debugFrameWasSentToEngine': is not true.


Either the assertion indicates an error in the framework itself, or we should provide substantially more information in this error message to help you determine and fix the underlying cause.
In either case, please report this assertion by filing a bug on GitHub:
  https://github.com/flutter/flutter/issues/new?template=02_bug.yml

When the exception was thrown, this was the stack: 
#2      WidgetsBinding.drawFrame.<anonymous closure> (package:flutter/src/widgets/binding.dart:1218:16)
#3      SchedulerBinding._executeTimingsCallbacks (package:flutter/src/scheduler/binding.dart:345:19)
#4      _invoke1 (dart:ui/hooks.dart:347:13)
#5      PlatformDispatcher._reportTimings (dart:ui/platform_dispatcher.dart:645:5)
#6      _reportTimings (dart:ui/hooks.dart:298:31)
(elided 2 frames from class _AssertionError)
The TimingsCallback that gets executed was: Closure: (List<FrameTiming>) => void
====================================================================================================

======== Exception caught by Flutter framework =====================================================
The following assertion was thrown while executing callbacks for FrameTiming:
'package:flutter/src/widgets/binding.dart': Failed assertion: line 1218 pos 16: 'debugFrameWasSentToEngine': is not true.


Either the assertion indicates an error in the framework itself, or we should provide substantially more information in this error message to help you determine and fix the underlying cause.
In either case, please report this assertion by filing a bug on GitHub:
  https://github.com/flutter/flutter/issues/new?template=02_bug.yml

When the exception was thrown, this was the stack: 
#2      WidgetsBinding.drawFrame.<anonymous closure> (package:flutter/src/widgets/binding.dart:1218:16)
#3      SchedulerBinding._executeTimingsCallbacks (package:flutter/src/scheduler/binding.dart:345:19)
#4      _invoke1 (dart:ui/hooks.dart:347:13)
#5      PlatformDispatcher._reportTimings (dart:ui/platform_dispatcher.dart:645:5)
#6      _reportTimings (dart:ui/hooks.dart:298:31)
(elided 2 frames from class _AssertionError)
The TimingsCallback that gets executed was: Closure: (List<FrameTiming>) => void
====================================================================================================

======== Exception caught by Flutter framework =====================================================
The following assertion was thrown while executing callbacks for FrameTiming:
'package:flutter/src/widgets/binding.dart': Failed assertion: line 1218 pos 16: 'debugFrameWasSentToEngine': is not true.


Either the assertion indicates an error in the framework itself, or we should provide substantially more information in this error message to help you determine and fix the underlying cause.
In either case, please report this assertion by filing a bug on GitHub:
  https://github.com/flutter/flutter/issues/new?template=02_bug.yml

When the exception was thrown, this was the stack: 
#2      WidgetsBinding.drawFrame.<anonymous closure> (package:flutter/src/widgets/binding.dart:1218:16)
#3      SchedulerBinding._executeTimingsCallbacks (package:flutter/src/scheduler/binding.dart:345:19)
#4      _invoke1 (dart:ui/hooks.dart:347:13)
#5      PlatformDispatcher._reportTimings (dart:ui/platform_dispatcher.dart:645:5)
#6      _reportTimings (dart:ui/hooks.dart:298:31)
(elided 2 frames from class _AssertionError)
The TimingsCallback that gets executed was: Closure: (List<FrameTiming>) => void
====================================================================================================

======== Exception caught by scheduler library =====================================================
The following assertion was thrown during a scheduler callback:
'package:flutter/src/semantics/semantics.dart': Failed assertion: line 4221 pos 14: 'node.parent?._dirty != true': is not true.


Either the assertion indicates an error in the framework itself, or we should provide substantially more information in this error message to help you determine and fix the underlying cause.
In either case, please report this assertion by filing a bug on GitHub:
  https://github.com/flutter/flutter/issues/new?template=02_bug.yml

When the exception was thrown, this was the stack: 
#2      SemanticsOwner.sendSemanticsUpdate (package:flutter/src/semantics/semantics.dart:4221:14)
#3      PipelineOwner.flushSemantics (package:flutter/src/rendering/object.dart:1502:24)
#4      PipelineOwner.flushSemantics (package:flutter/src/rendering/object.dart:1504:15)
#5      RendererBinding.drawFrame (package:flutter/src/rendering/binding.dart:636:25)
#6      WidgetsBinding.drawFrame (package:flutter/src/widgets/binding.dart:1242:13)
#7      RendererBinding._handlePersistentFrameCallback (package:flutter/src/rendering/binding.dart:495:5)
#8      SchedulerBinding._invokeFrameCallback (package:flutter/src/scheduler/binding.dart:1438:15)
#9      SchedulerBinding.handleDrawFrame (package:flutter/src/scheduler/binding.dart:1351:9)
#10     SchedulerBinding._handleDrawFrame (package:flutter/src/scheduler/binding.dart:1204:5)
#11     _invoke (dart:ui/hooks.dart:331:13)
#12     PlatformDispatcher._drawFrame (dart:ui/platform_dispatcher.dart:444:5)
#13     _drawFrame (dart:ui/hooks.dart:303:31)
(elided 2 frames from class _AssertionError)
====================================================================================================

======== Exception caught by Flutter framework =====================================================
The following assertion was thrown while executing callbacks for FrameTiming:
'package:flutter/src/widgets/binding.dart': Failed assertion: line 1218 pos 16: 'debugFrameWasSentToEngine': is not true.


Either the assertion indicates an error in the framework itself, or we should provide substantially more information in this error message to help you determine and fix the underlying cause.
In either case, please report this assertion by filing a bug on GitHub:
  https://github.com/flutter/flutter/issues/new?template=02_bug.yml

When the exception was thrown, this was the stack: 
#2      WidgetsBinding.drawFrame.<anonymous closure> (package:flutter/src/widgets/binding.dart:1218:16)
#3      SchedulerBinding._executeTimingsCallbacks (package:flutter/src/scheduler/binding.dart:345:19)
#4      _invoke1 (dart:ui/hooks.dart:347:13)
#5      PlatformDispatcher._reportTimings (dart:ui/platform_dispatcher.dart:645:5)
#6      _reportTimings (dart:ui/hooks.dart:298:31)
(elided 2 frames from class _AssertionError)
The TimingsCallback that gets executed was: Closure: (List<FrameTiming>) => void
====================================================================================================

```

</details>


```
stable : 3.32.0
master : 3.33.0-1.0.pre.138
```

**Comment by chunhtai** [comment]
requesting cp https://github.com/flutter/flutter/pull/169362
